src/lava/magma/core/model/nc/model.py

```diff
@@ -38,3 +102,146 @@ def allocate(self, net: Net):
         Note: This should work as before.
         """
         pass
+
+
+class NcProcessModel(AbstractNcProcessModel):
```

Comment:
Unclear what this class is for. An NcProcModel is for allocating resources on a NeuroCore. Execution of that ProcModel means some neuro cores will run according to the resource configurations requested by this ProcModel. Why do we need a Python implementation for running an NcProcModel with all kinds of channel management?

The real NeuroCoreProcess running on a neuro core will communicate with NxRuntime for pausing and stopping. If any SeqProc wants to interact with Vars, it will do so directly. This does not have to go through the NxSdkRuntimeService. If the user wants to manipulate Vars at runtime, the most natural way would be for the NxSDKRuntimeservice to do this via the Board directly.

Reply:
Seems like there is overall architectural documentation for the overall usage modal and why this is necessary.